collector/bgp.go

```diff
@@ -340,6 +348,37 @@ func getPeerAdvertisedPrefixes(ch chan<- prometheus.Metric, wg *sync.WaitGroup,
 	newGauge(ch, bgpDesc["prefixAdvertisedCount"], float64(advertisedPrefixes.TotalPrefixCounter), peerLabels...)
 }
 
+type bgpRoutes struct {
+	// We care only about the routes
+	Routes map[string][]json.RawMessage `json:"routes"`
+}
+
+func getPeerAcceptedFilteredRoutes(ch chan<- prometheus.Metric, wg *sync.WaitGroup, AFI string, SAFI string, vrfName string, neighbor string, prefixesReceived float64, logger *slog.Logger, bgpDesc map[string]*prometheus.Desc, peerLabels ...string) {
+	var cmd string
+	if strings.ToLower(vrfName) == "default" {
+		cmd = fmt.Sprintf("show bgp  %s %s neighbors %s routes json", strings.ToLower(AFI), strings.ToLower(SAFI), neighbor)
+	} else {
+		cmd = fmt.Sprintf("show bgp vrf %s %s %s neighbors %s routes json", vrfName, strings.ToLower(AFI), strings.ToLower(SAFI), neighbor)
+	}
+
+	output, err := executeBGPCommand(cmd)
+	if err != nil {
+		logger.Error("get neighbor accepted filtered routes failed", "afi", AFI, "safi", SAFI, "vrf", vrfName, "neighbor", neighbor, "err", err)
+		return
+	}
+
+	var routes bgpRoutes
+	if err := json.Unmarshal(output, &routes); err != nil {
+		logger.Error("get neighbor accepted filtered routes failed", "afi", AFI, "safi", SAFI, "vrf", vrfName, "neighbor", neighbor, "err", err)
+		return
+	}
+
+	prefixesAccepted := float64(len(routes.Routes))
```

Comment:
Do you think it's worth adding some validation around this?

Reply:
I don't think so because `len` returns `0` if `routes.Routes is `nil` or empty. Converting `0` to a `float64` results in `0.0`.